src/openms/source/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.cpp

```diff
@@ -103,41 +103,53 @@ namespace OpenMS
   }
 
   void MRMTransitionGroupPicker::calculateBgEstimation_(const MSChromatogram& chromatogram,
-      double best_left, double best_right, double & background, double & avg_noise_level)
+      double best_left, double best_right, double peak_height, double & background, double & avg_noise_level)
   {
     // determine (in the chromatogram) the intensity at the left / right border
-    MSChromatogram::const_iterator it = chromatogram.begin();
-    int nr_points = 0;
-    for (; it != chromatogram.end(); ++it)
+    double intensity_left = 0.0;
+    double rt_apex = 0.0;
+    double intensity_right = 0.0;
+    for (MSChromatogram::const_iterator it = chromatogram.begin() + 1; it != chromatogram.end(); it++)
     {
-      if (it->getMZ() > best_left)
+      MSChromatogram::const_iterator it_prev = it;
+      it_prev--; //previous point
+
+      if (it->getMZ() == best_left)
       {
-        nr_points++;
-        break;
+        intensity_left = it->getIntensity();
       }
-    }
-    double intensity_left = it->getIntensity();
-    for (; it != chromatogram.end(); ++it)
-    {
-      if (it->getMZ() > best_right)
+      else if (it->getIntensity() >= peak_height && it_prev->getIntensity() <= peak_height)
+      {
+        rt_apex = (it->getMZ() + it_prev->getMZ())/2;
+      }
+      else if (it->getMZ() == best_right)
       {
-        break;
+        intensity_right = it->getIntensity();
       }
-      nr_points++;
     }
-    if (it == chromatogram.begin() || nr_points < 1)
+
+    double intensity_max, intensity_min, rt_max, rt_min;
+    if (intensity_left >= intensity_right)
     {
-      // something is fishy, the endpoint of the peak is the beginning of the chromatogram
-      std::cerr << "Tried to calculate background but no points were found " << std::endl;
-      return;
+      intensity_max = intensity_left;
+      rt_max = best_left;
+      intensity_min = intensity_right;
+      rt_min = best_right;
     }
+    else 
+    {
+      intensity_max = intensity_right;
+      rt_max = best_right;
+      intensity_min = intensity_left;
+      rt_min = best_left;
+    }
+    double delta_int = intensity_max - intensity_min;
+    double delta_rt = best_right - best_left;
+    double delta_rt_apex = abs(rt_max-rt_apex);
+    double delta_int_apex = delta_int*delta_rt_apex/delta_rt;
 
-    // decrease the iterator and the nr_points by one (because we went one too far)
-    double intensity_right = (it--)->getIntensity();
-    nr_points--;
-
-    avg_noise_level = (intensity_right + intensity_left) / 2;
-    background = avg_noise_level * nr_points;
+    avg_noise_level = intensity_max - delta_int_apex;
+    background = intensity_min*delta_rt + 0.5*delta_int*delta_rt;
```

Comment:
it seems that here you are integrating using the real RT. This however wont work with the rest of the code which basically uses "measurement points" and does not use RT. Basically the rest of the code sums up intensity, so we cannot start using `delta_rt` here or the background estimate will be completely off.